src/packerlicious/packer.py

```diff
@@ -0,0 +1,73 @@
+import subprocess
+import shlex
+import os
+import tempfile
+from packerlicious.template import Template
+
+class Packer(object):
+    """
+    Packer Binary Wrapper
+    """
+
+    packer_path = ""
+    template = None
+    _vars = {}
+
+    def __init__(self, template=None, _vars={}, packer_path=""):
+        if len(packer_path) > 0 and packer_path[-1] != '/':
+            packer_path += "/"
+        self.packer_path = packer_path
+        self.template = template
+        self._vars = _vars
+        self.check_available()
+
+    def check_available(self):
+        """
+        Check if packer binary is available.
+        Raises ValueError if not.
+        """
+        try:
+            subprocess.Popen(self.packer_path + "packer", stdout=subprocess.PIPE)
+        except OSError:
+            raise ValueError("Packer Binary not available...")
+
+    def add_template(self, template):
+        self.template = template
+    
+    def add_var(self, var):
+        self._vars.update(var)
+    
+    def add_vars(self, _vars):
+        self._vars = _vars
+
+    def validate(self, syntax_only=False, _except=None, only=None):
+        """
+        Validate template.
+        Returns (Return Code, Output)
+        """
+        self.check_available()
+        if self.template is None:
+            raise ValueError("No template provided...")
+        validate_command = self.packer_path + "packer validate "
+        if syntax_only:
+            validate_command += "-syntax_only "
+        if _except is not None:
+            validate_command += "-except="
+            for e in _except:
+                validate_command += e + ","
+            validate_command += " "
+        if only is not None:
+            validate_command += "-only="
+            for o in only:
+                validate_command += o + ","
+            validate_command += " "
+        for key in self._vars:
+            validate_command += "-var '" + key + "=" + self._vars[key] + "' "
+        template_file, template_filename = tempfile.mkstemp()
+        os.write(template_file, self.template.to_json())
```

Comment:
ideally this would pipe directly to packer without creating a tmp file.
if you keep the tmp file writing, please add error handling

if you don't change it on this pr, please create an issue about this so the idea doesn't get lost.

Reply:
I did not find a way to pipe it directly without creating file.
If you have a way, plz tell me so I can update accordingly.

Reply:
may be able to do it via stdin of the `popen.communicate`. see https://github.com/hashicorp/packer/issues/2914 for an example. looks like there is a trailing `-` they pass to the cmd.

Also more info here https://github.com/hashicorp/packer/issues/4132

Reply:
Yes, did it :+1: 
Thanks :)